packages/lib-classifier/src/store/TutorialStore/TutorialStore.js

```diff
@@ -109,31 +84,33 @@ const TutorialStore = types
   }))
 
   .actions(self => {
-    function afterAttach () {
-      createWorkflowObserver()
-      createUPPObserver()
+    function _onUPPLoaded() {
+      const upp = getRoot(self).userProjectPreferences
+      if (upp.loadingState === asyncStates.success) {
+        self.showTutorialInModal()
+      }
     }
 
-    function createWorkflowObserver () {
-      const workflowDisposer = autorun(() => {
-        const workflow = tryReference(() => getRoot(self).workflows.active)
-        if (workflow) {
-          self.reset()
-          self.resetSeen()
-          self.fetchTutorials()
-        }
-      }, { name: 'Tutorial Store Workflow Observer autorun' })
-      addDisposer(self, workflowDisposer)
+    function _onTutorialChange() {
+      const workflow = tryReference(() => getRoot(self).workflows.active)
+      if (workflow) {
+        const { tutorial } = workflow
+        self.setActiveTutorial(tutorial?.id)
+      }
```

Comment:
Good point. I think this should observe `workflow?.tutorial`, or maybe move to the `Workflow` model, where it can observe `self.tutorial`.

Reply:
On second thoughts, maybe we can get rid of this side effect altogether. Components that use `tutorials.active` could use `workflows.active?.tutorial` instead. 

I think that would remove the need to keep `tutorials.active` in sync with the workflow store, which is a potential source of bugs.

Reply:
I'm going to refactor `ModalTutorial` to remove `inject and to replace `tutorials.active` with `workflows.active?.tutorial`. I think that will mean this reaction can be deleted. I feel more confident about the code if views are observing `workflows.active` directly rather than waiting for `workflows.active` to trigger an update to `tutorials.active`, which then updates the displayed tutorial.

Reply:
This turned out to be a bigger refactor than I thought, but I've pushed a commit that removes `setActiveTutorial` from the tutorials store, in favour of using `workflows.active?.tutorial` instead.

It turns out that `TutorialStore` is also responsible for managing the active tutorial step, which made that change more complex than I thought it would be. It should all be working now, but I'm noticing that I get the popup, even on a second visit.